### PR TITLE
Remove istiod.enabled 

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: clyang82 <clyang@cn.ibm.com>

Remove istiod.enabled which is not used anymore.

### CODE REVIEW

File: operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml [hunk 1]
```diff
@@ -624,7 +624,7 @@ spec:
             - --proxyAdminPort
             - "15000"
             - --controlPlaneAuthPolicy
-            - NONE
+            - MUTUAL_TLS
```
Comment (howardjohn) [inline operator/cmd/mesh/testdata/manifest-generate/output/all_on.golden-show-in-gh-pull-request.yaml]: this does not look correct, it should be NONE. we probably need to remove this flag longer term

**Review by howardjohn — CHANGES_REQUESTED** [review]
can you revert the control plane auth part?

**Comment by clyang82** [comment]
> can you revert the control plane auth part?

@howardjohn from the code changes, you will see the `istiod.enable` is only used by `prometheus` deployment. so I think `istiod.enabled` can be able to remove. anyway, let me double check today. Thanks.

**Comment by clyang82** [comment]
@howardjohn I just removed the istiod.enabled in 
```
global.yaml
operator/data/profiles/default.yaml
operator/data/profiles/separate.yaml
manifests/istio-telemetry/prometheus/templates/deployment.yaml
```
then run `make gen`, it updates the others automatically. What do you mean `revert the control plane auth part`?

**Comment by howardjohn** [comment]
If you are just changing a value that was always true, then the generated output should not change. But in this case it did change

**Pull request closed without merging**